tests/bluetooth/iso/modules/iso_broadcast_sink.c

```diff
@@ -0,0 +1,535 @@
+			err = bt_le_per_adv_sync_delete(sync);
+			if (err) {
+				LOG_INF("failed (err %d)", err);
+				return;
+			}
+			continue;
+		}
+		LOG_INF("Periodic sync established.");
+
+big_sync_create:
+		LOG_INF("Create BIG Sync...");
+		err = bt_iso_big_sync(sync, &big_sync_param, &big);
+		if (err) {
+			LOG_INF("failed (err %d)", err);
+			return;
+		}
+		LOG_INF("success.");
+
+		for (uint8_t chan = 0U; chan < big_sync_param.num_bis; chan++) {
+			LOG_INF("Waiting for BIG sync chan %u...", chan);
+			err = k_sem_take(&sem_big_sync, TIMEOUT_SYNC_CREATE);
+			if (err) {
+				break;
+			}
+			LOG_INF("BIG sync chan %u successful.", chan);
+		}
+		if (err) {
+			LOG_INF("failed (err %d)", err);
+
+			LOG_INF("BIG Sync Terminate...");
+			err = bt_iso_big_terminate(big);
+			if (err) {
+				LOG_INF("failed (err %d)", err);
+				return;
+			}
+			LOG_INF("done.");
+
+			goto per_sync_lost_check;
+		}
+		LOG_INF("BIG sync established.");
+
+		for (uint8_t chan = 0U; chan < big_sync_param.num_bis; chan++) {
+			LOG_INF("Waiting for BIG sync lost chan %u...", chan);
+			err = k_sem_take(&sem_big_sync_lost, K_FOREVER);
+			if (err) {
+				LOG_INF("failed (err %d)", err);
+				return;
+			}
+			LOG_INF("BIG sync lost chan %u.", chan);
+		}
+		LOG_INF("BIG sync lost.");
+
+per_sync_lost_check:
+		err = k_sem_take(&sem_per_sync_lost, K_NO_WAIT);
+		if (err) {
+			/* Periodic Sync active, go back to creating BIG Sync */
+			goto big_sync_create;
+		}
+		LOG_INF("Periodic sync lost.");
+	}
+}
+
+int iso_broadcast_sink_init(void)
+{
+	running = false;
+
+	if (!gpio_is_ready_dt(&led)) {
+		return -EBUSY;
+	}
+
+	k_thread_create(&broadcaster_thread, broadcaster_sink_thread_stack,
+			K_THREAD_STACK_SIZEOF(broadcaster_sink_thread_stack), broadcaster_t, NULL,
+			NULL, NULL, 5, K_USER, K_NO_WAIT);
+
+	return 0;
+}
+
+int iso_broadcast_sink_start(const struct shell *shell, size_t argc, char **argv)
+{
+	int ret;
+
+	ret = gpio_pin_configure_dt(&led, GPIO_OUTPUT_ACTIVE);
+	if (ret < 0) {
+		return ret;
+	}
+
+	running = true;
+	return 0;
+}
+
+static int argument_check(const struct shell *shell, uint8_t const *const input)
+{
+	char *end;
+	int arg_val = strtol(input, &end, 10);
+
+	if (*end != '\0' || (uint8_t *)end == input || (arg_val == 0 && !isdigit(input[0])) ||
+	    arg_val < 0) {
+		shell_error(shell, "Argument must be a positive integer %s", input);
+		return -EINVAL;
+	}
+
+	if (running) {
+		shell_error(shell, "Arguments can not be changed while running");
+		return -EACCES;
+	}
+
+	return arg_val;
+}
+
+static struct option long_options[] = {
+					{ "num_bis:", required_argument, NULL, 'n' },
+					{ 0, 0, 0, 0 } };
+
+static const char short_options[] = "n:";
+
+static int set_param(const struct shell *shell, size_t argc, char **argv)
+{
+	int result = argument_check(shell, argv[2]);
+
+	if (result < 0) {
+		return result;
+	}
+
+	if (running) {
+		shell_error(shell, "Change sink parameters before starting");
```

Comment:
Is there any major reason, except the additional cleanup and handling, why receiving data cannot be stopped once started?